packages/api/tests/lib/test-granules.js

```diff
@@ -1,12 +1,206 @@
 const test = require('ava');
+const cryptoRandomString = require('crypto-random-string');
+
+const { RecordDoesNotExist } = require('@cumulus/errors');
+
+const {
+  CollectionPgModel,
+  FilePgModel,
+  generateLocalTestDb,
+  GranulePgModel,
+  localStackConnectionEnv,
+} = require('@cumulus/db');
+
+// PG mock data factories
+const {
+  fakeCollectionRecordFactory,
+  fakeGranuleRecordFactory,
+} = require('@cumulus/db/dist/test-utils');
 
 const {
+  createBucket,
+  createS3Buckets,
+  deleteS3Buckets,
+  s3ObjectExists,
+  s3PutObject,
+} = require('@cumulus/aws-client/S3');
+
+const { randomId, randomString } = require('@cumulus/common/test-utils');
+
+const models = require('../../models');
+
+// Dynamo mock data factories
+const {
+  fakeCollectionFactory,
+  fakeGranuleFactoryV2,
+} = require('../../lib/testUtils');
+
+const {
+  deleteGranuleAndFiles,
   getExecutionProcessingTimeInfo,
+  getGranuleProductVolume,
   getGranuleTimeToArchive,
   getGranuleTimeToPreprocess,
-  getGranuleProductVolume,
 } = require('../../lib/granules');
 
+const { migrationDir } = require('../../../../lambdas/db-migration');
+
+const testDbName = `granules_${cryptoRandomString({ length: 10 })}`;
+
+const collectionId = 456;
+
+let collectionModel;
+let filePgModel;
+let granuleModel;
+let granulePgModel;
+let s3Buckets = {};
+
+process.env.CollectionsTable = randomId('collection');
+process.env.GranulesTable = randomId('granules');
+process.env.stackName = randomId('stackname');
+process.env.system_bucket = randomId('systembucket');
+process.env.TOKEN_SECRET = randomId('secret');
+
+/**
+ * Helper for creating a granule, and files belonging to that granule
+ * @param {Knex} dbClient - Knex client
+ * @param {number} collectionCumulusId - cumulus_id for the granule's parent collection
+ * @param {boolean} published - if the granule should be marked published to CMR
+ * @returns {Object} PG and Dynamo granules, files
+ */
+async function createGranuleAndFiles(dbClient, collectionCumulusId, published) {
+  s3Buckets = {
```

Comment:
I would initialize this variable inside this function, rather than in global scope, and return the value from the function so that tests using it for setup can teardown the resources used for the test

Reply:
Yup, so in the next PR I have a refactor on this function to move it to a shared space and imported in each of these tests. Right now there's a lot of duplication. Just FYI.

Here are the changes to make this consistent with the other tests and clean up the buckets properly in this file (previously I was relying on an `.after` hook):

https://github.com/nasa/cumulus/pull/2057/commits/6be202d3416e94f572e9b25d94b40034e0fd0500